pkg/controller/featuregates_hook_test.go

```diff
@@ -257,9 +257,10 @@ func TestUpdateDeploymentFeatureGatesHook(t *testing.T) {
 		}
 		mockMapper.ValidateCalls(t, 1, 1, 0, 0)
 
+		// The new code will now validate that the arguments are the same, and accept that as valid
 		err = update(nil, dep)
-		if err == nil {
-			t.Fatal("no error in second update")
+		if err != nil {
+			t.Fatal("error in second update")
 		}
```

Comment:
Seeing a bunch of `err == nil` flip to `err != nil` is making my spidey sense go off.

The test scenarios seemed to originally assume that an error _would_ occur. Now, because of the code changes, the scenarios mean that an error does not occur. To me, it feels like the appropriate thing to do is change the test inputs so that the test assertions continue to hold, and then add some new tests to cover the various success/error outcomes we expect from the new code.

Reply:
Yeah, running the update a second time is now a no-op, so perhaps it makes sense to remove it.